src/converter/one/jfr/Index.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.jfr;
+
+/**
+ * Fast and compact Object->int map.
+ */
+public class Index<T> {
+    protected static final int INITIAL_CAPACITY = 16 * 1024;
+
+    private Object[] keys;
+    private int[] values;
+
+    public int size;
+
+    public Index() {
+        this(INITIAL_CAPACITY);
+    }
+
+    public Index(int initialCapacity) {
+        this.keys = new Object[initialCapacity];
+        this.values = new int[initialCapacity];
+    }
+
+    public int index(T key) {
+        if (key == null) {
+            throw new NullPointerException("Null key is not allowed");
+        }
+
+        int mask = keys.length - 1;
+        int i = hashCode(key) & mask;
+        while (true) {
+            Object currentKey = keys[i];
+            if (currentKey == null) {
+                break;
+            }
+            //noinspection unchecked
```

Comment:
This is an IDE-specific hint. `javac` will still issue a warning. Please use `@SuppressWarnings` instead.

Reply:
moreover, it is unneeded in current implementation, fixed